### PR TITLE
feat(asdf): Add rebar3 to asdf manager

### PR DESCRIPTION
**Pull request by kivra-pauoli**
## Changes

This pull request adds support to bumping `rebar` entries in `.tool-versions` files, as supported by https://asdf-vm.com/.

## Context

I saw `rebar` was not supported, and the change seems small enough for a first pull request.

## Further considerations

Mind you that `rebar3` is declared as `rebar` in `.tool-versions` but fetched from [erlang/rebar3](https://github.com/erlang/rebar3).

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by kivra-pauoli** [comment]
As per the comments in the pull request template I tried to sign the CLA in link https://cla-assistant.io/renovatebot/renovate, but I'm met with a mostly empty page. It's possible that approving the CI runs will show a different input.

image

**Review by rarkins — DISMISSED** [review]

**Comment by rarkins** [comment]
@kivra-pauoli thanks for your PR and welcome to the repo!

File: lib/modules/manager/asdf/extract.spec.ts [hunk 1]
```diff
@@ -484,6 +485,13 @@ dummy 1.2.3
             depName: 'python',
             extractVersion: '^v(?<version>\\S+)',
           },
+          {
+            currentValue: '3.23.0',
+            datasource: 'github-tags',
+            packageName: 'erlang/rebar3',
+            depName: 'rebar',
+            extractVersion: '^(?<version>\\S+)',
```
Comment (viceice) [inline lib/modules/manager/asdf/extract.spec.ts:493]:
```suggestion
```

not needed

File: lib/modules/manager/asdf/upgradeable-tooling.ts [hunk 2]
```diff
@@ -500,6 +500,14 @@ export const upgradeableTooling: Record<string, ToolingDefinition> = {
       extractVersion: '^v(?<version>\\S+)',
     },
   },
+  rebar: {
+    asdfPluginUrl: 'https://github.com/Stratus3D/asdf-rebar.git',
+    config: {
+      datasource: GithubTagsDatasource.id,
+      packageName: 'erlang/rebar3',
+      extractVersion: '^(?<version>\\S+)',
```
Comment (viceice) [inline lib/modules/manager/asdf/upgradeable-tooling.ts:508]:
```suggestion
```
see other comment

File: lib/modules/manager/asdf/extract.spec.ts (hunk 1 above)
Comment (kivra-pauoli) [reply lib/modules/manager/asdf/extract.spec.ts:493]: 👍 fwiw, there's 35 other matches similar to this one, which is what lead me in "error".

Comment (kivra-pauoli) [reply lib/modules/manager/asdf/extract.spec.ts]: 7794cae75

File: lib/modules/manager/asdf/upgradeable-tooling.ts (hunk 2 above)
Comment (kivra-pauoli) [reply lib/modules/manager/asdf/upgradeable-tooling.ts]: 7794cae75

File: lib/modules/manager/asdf/extract.spec.ts (hunk 1 above)
Comment (rarkins) [reply lib/modules/manager/asdf/extract.spec.ts]: It depends on how the repo tags its releases. rebar doesn't use v: https://github.com/erlang/rebar3/tags

**Review by viceice — APPROVED** [review]

**Pull request merged**